validator/src/cli/thread_args.rs

```diff
@@ -0,0 +1,42 @@
+//! Arguments for controlling the number of threads allocated for various tasks
+
+use {
+    clap::Arg, solana_clap_utils::input_validators::is_within_range,
+    solana_rayon_threadlimit::get_max_thread_count,
+};
+
+pub struct DefaultThreadArgs {
+    pub replay_forks_threads: String,
+    pub replay_transactions_threads: String,
+}
+
+impl Default for DefaultThreadArgs {
+    fn default() -> Self {
+        let num_total_threads = get_max_thread_count();
+
+        Self {
+            replay_forks_threads: 1.to_string(),
+            replay_transactions_threads: num_total_threads.to_string(),
+        }
+    }
+}
+
+pub fn thread_args<'a>(defaults: &DefaultThreadArgs) -> Vec<Arg<'_, 'a>> {
+    // Do not let any threadpool size scale over the number of threads
+    vec![
+        Arg::with_name("replay_forks_threads")
+            .long("replay-forks-threads")
+            .takes_value(true)
+            .value_name("NUMBER")
+            .default_value(&defaults.replay_forks_threads)
+            .validator(|num| is_within_range(num, 1..=get_max_thread_count()))
+            .help("Number of threads to use for replay of blocks on different forks"),
+        Arg::with_name("replay_transactions_threads")
+            .long("replay-transactions-threads")
+            .takes_value(true)
+            .value_name("NUMBER")
+            .default_value(&defaults.replay_transactions_threads)
+            .validator(|num| is_within_range(num, 1..=get_max_thread_count()))
```

Comment:
I went back and forth on whether to leave this visible or to hide them with `.hidden(hidden_unless_forced())`; ended up leaving visible, but given that these are advanced and brand new, maybe we should keep them hidden ?

Reply:
I'm neutral, but I know how our boy Trent feels about foot guns

Reply:
Yeah, I made visible but after sleeping on it, thinking of going back towards hidden. These are advanced options, and mostly to allow us to tune without having to rebuild a binary. Ideally, we fine a heuristic that is ideal for every machine, and nobody ever has to touch these

Reply:
any arg should be hidden until we're ready to offer support for the logic it enables